### PR TITLE
Check string to replace before running the replace

### LINKED ISSUE #4149: Replacing an empty search field hangs GDevelop
A forum user [reports](https://forum.gdevelop.io/t/search-replace-problem/41075) the following issue, that I confirmed on my side.

Reproduction steps:
Open any project,
Go to any events sheet,
Press Ctrl+f to open the search, 
Press Tab to access the Replace field,
Type anything,
Hit Return.

GDevelop 5.0.139 hangs indefinitely.

### PR DESCRIPTION
**Pull request by AlexandreSi**
Fixes #4149

### CODE REVIEW

File: Core/GDCore/IDE/Events/EventsRefactorer.cpp [hunk 1]
```diff
@@ -565,6 +565,8 @@ std::vector<EventsSearchResult> EventsRefactorer::ReplaceStringInEvents(
     bool inActions,
     bool inEventStrings) {
   vector<EventsSearchResult> modifiedEvents;
+  if (toReplace.empty()) return modifiedEvents;
```
Comment (fannieyan) [inline Core/GDCore/IDE/Events/EventsRefactorer.cpp:568]: Should it test newString.empty() too ? (I wonder if replacing a string with an empty one is a use case?)

**Review by fannieyan — COMMENTED** [review]
👍

**Review by D8H — APPROVED** [review]
LGTM

File: Core/GDCore/IDE/Events/EventsRefactorer.cpp (hunk 1 above)
Comment (D8H) [reply Core/GDCore/IDE/Events/EventsRefactorer.cpp:568]: I use this in text editors to remove things.

**Review by 4ian — APPROVED** [review]

**Pull request merged**